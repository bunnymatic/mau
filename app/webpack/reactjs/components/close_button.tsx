import React, { FC } from "react";

interface CloseButtonProps {
  handleClick: () => void;
}

export const CloseButton: FC<CloseButtonProps> = ({ handleClick }) => {
  return (
    <a
      className="close-button"
      title="close"
      href="#"
      onClick={(ev) => {
        ev.preventDefault();
        handleClick();
      }}
    >
      <i className="fa fa-times"></i>
    </a>
  );
};
