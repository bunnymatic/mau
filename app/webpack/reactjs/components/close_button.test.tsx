import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { beforeEach, describe, expect, it, vi } from "vitest";

import { CloseButton } from "./close_button";

describe("CloseButton", () => {
  let mockHandleClose = vi.fn();
  let container;

  beforeEach(() => {
    vi.resetAllMocks();
    const rendered = render(<CloseButton handleClick={mockHandleClose} />);
    container = rendered.container;
  });

  it("matches the snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("runs handle close when you click close", () => {
    const button = screen.getByTitle("close");
    fireEvent.click(button);
    expect(mockHandleClose).toHaveBeenCalled();
  });
});
