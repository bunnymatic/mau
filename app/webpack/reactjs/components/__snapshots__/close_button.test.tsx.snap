// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CloseButton > matches the snapshot 1`] = `
<div>
  <a
    class="close-button"
    href="#"
    title="close"
  >
    <i
      class="fa fa-times"
    />
  </a>
</div>
`;
